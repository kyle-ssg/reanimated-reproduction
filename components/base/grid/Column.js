import React, {PureComponent} from 'react';
import propTypes from 'prop-types';
import cn from 'classnames';

const _propTypes = {
    /** The element's children */
    children: propTypes.node,
    /** The element's class name */
    className: propTypes.string,
};

/**
 * Div with standard horizontal padding
 */
export class Column extends PureComponent {
    static displayName = 'Column';

    static propTypes = _propTypes;

    render() {
        const {props: {className, ...rest}} = this;
        return (
            <div {...rest} className={cn(className, 'flex-column')}/>
        );
    }
}

global.Column = Column;
export default Column;
