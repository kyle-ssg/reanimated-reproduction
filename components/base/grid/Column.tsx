import cn from 'classnames'
import { FC } from 'react'

interface Column {
  className?: string
}

//Div with standard horizontal padding
const Column: FC<Column> = ({ className, children, ...props }) => (
  <div {...props} className={cn(className, 'flex-column')}>
    {children}
  </div>
)

Column.displayName = 'Column'
export default Column
