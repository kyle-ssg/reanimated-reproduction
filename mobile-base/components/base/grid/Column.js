/**
 * Created by Kyle on 17/06/2016.
 */
module.exports = (props) => <View style={[Styles.column, props.style]}>{props.children}</View>