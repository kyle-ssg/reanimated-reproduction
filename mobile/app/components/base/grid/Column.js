/**
 * Created by Kyle on 17/06/2016.
 */
import React from "react";
import propTypes from "prop-types";
import Flex from "./Flex";

const Column = (props) => (
  <View style={[Styles.column, props.style]}>{props.children}</View>
);

const FCol = (props) => (
  <Flex style={props.flexStyle}>
    <Column style={[Styles.column, props.style]}>{props.children}</Column>
  </Flex>
);

Column.propTypes = FCol.propTypes = {
  flexStyle: propTypes.any,
  style: propTypes.any,
  children: propTypes.any,
};

module.exports = { Column, FCol };
