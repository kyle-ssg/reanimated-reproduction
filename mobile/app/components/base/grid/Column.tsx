import React from 'react'
import Flex from './Flex'

export type ColumnType = {
  flexStyle?: any
  style?: ReactNative.ViewStyle
  children?: React.ReactNode
}

export const Column: React.FC<ColumnType> = ({ style, children }) => (
  <View style={[Styles.column, style]}>{children}</View>
)

export default Column
