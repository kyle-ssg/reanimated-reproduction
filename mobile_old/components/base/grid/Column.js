/**
 * Created by Kyle on 17/06/2016.
 */
const Column = (props) => (
  <Animated.View style={[Styles.column, props.style]}>{props.children}</Animated.View>
);

Column.propTypes = {
  style: React.PropTypes.any,
  children: OptionalObject
};

module.exports = Column;
