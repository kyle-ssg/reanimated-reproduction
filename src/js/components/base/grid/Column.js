//propTypes: value: OptionalNumber

const Flex = class extends React.Component {
	render() {
		return (
			<div {... this.props} className={(this.props.className || '') + ' flex-column'}>
				{this.props.children}
			</div>
		);
	}
};

Flex.defaultProps = {
	value: 1
};

Flex.propTypes = {
	className: OptionalString,
	value: OptionalNumber,
	children: OptionalNode,
	style: React.PropTypes.any
};

module.exports = Flex;
