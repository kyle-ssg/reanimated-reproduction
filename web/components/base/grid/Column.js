import React from 'react';
import cn from 'classnames';
import propTypes from 'prop-types';

const Column = ({ className, children, ...rest }) => (
    <div
      {...rest}
      className={cn({
          'flex-column': true,
      }, className)}
    >
        {children}
    </div>
);

Column.displayName = 'Column';

Column.defaultProps = {};

Column.propTypes = {
    className: propTypes.string,
    value: propTypes.number,
    children: propTypes.node,
    style: propTypes.any,
};

window.Column = Column;