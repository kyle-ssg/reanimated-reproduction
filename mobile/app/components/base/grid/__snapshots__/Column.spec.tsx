import * as React from 'react'
import Column from '../Column'
import renderer from 'react-test-renderer'

describe('<Column />', () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<Column />)
    expect(tree).toMatchSnapshot()
  })
})
