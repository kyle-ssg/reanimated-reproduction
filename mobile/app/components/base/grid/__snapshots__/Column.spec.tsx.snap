// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Column /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 10,
        "marginRight": 10,
      },
      undefined,
    ]
  }
/>
`;
